bson/_cbsonmodule.c

```diff
@@ -2797,6 +2816,85 @@ static PyObject* _cbson_decode_all(PyObject* self, PyObject* args) {
     return result;
 }
 
+
+static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* args) {
+    uint32_t size;
+    uint32_t value_length;
+    uint32_t position = 0;
+    buffer_t buffer;
+    const char* string;
+    PyObject* arr;
+    PyObject* result = NULL;
+    Py_buffer view = {0};
+
+    if (!PyArg_ParseTuple(args, "O", &arr)) {
+        return NULL;
+    }
+
+    buffer = pymongo_buffer_new();
+
+    if (!_get_buffer(arr, &view)) {
+        goto fail;
+    }
+
+    string = (char*)view.buf;
```

Comment:
Is .buf a pointer to the start view's memory or a pointer to the start of the original bytes() the view points to (ie. memoryview.obj in pure python)?

Reply:
This is the same approach we used in `_cbson_bson_to_dict`, as I understand it `.buf` in this case points to the underlying bytes in the `Py_Buffer` structure.